lib/ds/RequestExecutor.js

```diff
@@ -84,7 +84,7 @@ RequestExecutor.prototype.execute = function executeRequest(req, callback) {
     }
 
     if (response.statusCode > 399) {
-      return callback(new ResourceError(body || {status:response.statusCode}), null);
+      return callback(new ResourceError(body || {status:response.statusCode}, {url: req.uri, method: req.method}), null);
```

Comment:
Are you really meaning `url` here and not `uri`? Because the value that I received was `/tenants/current`. Which is not a URL.